### ISSUE TITLE
Cannot upload custom Forge build to local repo

### ISSUE THREAD

**Issue by maxanier** [MinecraftForge/MinecraftForge#3357]
Forge's build.gradle is actually prepared to upload built archives to a local repo folder ([code](https://github.com/MinecraftForge/MinecraftForge/blob/9a9c02fa2598f1f025993686789c9e3bc10385db/build.gradle#L240-L250)) if there is no pass for the official maven repo.
But that task fails no changelog file is generated (because no jenkins password is set).
There already is a fix for that in the code, but it is commented out.

[build.gradle#87](https://github.com/MinecraftForge/MinecraftForge/blob/9a9c02fa2598f1f025993686789c9e3bc10385db/build.gradle#L87)
```java
    // skip if there is no forge jenkins pass
    onlyIf {
        project.hasProperty('forgeJenkinsPass')
        // Not sure what abrar was trying to do here...
        //project.file("build/distributions/${project.name}-${project.version}-changelog.txt").text = ""
    }
```
If the last line is activated again, everything works again, and you can test your custom builds in your mods development environment.

For those interested:
Just add 
`    maven {
        name = "LocalForge"
        url = "../../MinecraftForge/repo/"
    }` to the repositories in your build.gradle.

**Comment by LexManos** [comment]
not sure what you're working with but the local building of the libs should be fine. I do it all the time.
gradlew uA tosses all the binaries into ./repo/

**Comment by maxanier** [comment]
Maybe you have the 'forgeJenkinsPass' in your local env variables or you have an 'old' changelog file in your build/distribution folder.

When running 'gradlew uploadArchives' I get:

> Execution failed for task ':uploadArchives'.
>  > Could not publish configuration 'archives'
>     > Cannot publish artifact 'forge-changelog.txt (net.minecraftforge:forge:1.10.2-12.18.2.0-custom_entity_selectors)' (/home/max/Dokumente/Programmieren/IntelliJ Minecraft/MinecraftForge/build/distributions/forge-1.10.2-12.18.2.0-custom_entity_selectors-changelog.txt) as it does not exist.

However if I uncomment the mentioned line, it generates the (empty) changelog.txt in the build/distributions folder and 'uploadArchives' succeeds.
(After it was generated once, it will use the old one, if no new one is generated).

I'm using the setup from cpw video and everything, but this, works like a charm.
Not sure if no one else uses uploadArchives or if they do not have this problem. (Although @AbrarSyed apparently had a similar problem).

**Comment by AbrarSyed** [comment]
could also find a way to just remove the changelog stuff entirely... its all in the buildscipt.

**Comment by maxanier** [comment]
Was fixed in c416f384613b5f7fa922c4cb6ad8e7137bbf520a, thx :)
